bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java

```diff
@@ -15,51 +15,103 @@
  */
 package com.google.cloud.bigtable.config;
 
-import java.io.Serializable;
-import java.util.Objects;
-
 import com.google.cloud.bigtable.grpc.BigtableInstanceName;
-import com.google.cloud.bigtable.grpc.BigtableSession;
+import com.google.auto.value.AutoValue;
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.base.MoreObjects;
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
+import java.io.Serializable;
+import javax.annotation.Nullable;
 
 /**
  * An immutable class providing access to configuration options for Bigtable.
  *
  * @author sduskis
  * @version $Id: $Id
  */
-//TODO: Perhaps break this down into smaller options objects?
-//TODO: This should be @Autovalue + Builder
-public class BigtableOptions implements Serializable, Cloneable {
+@AutoValue
+public abstract class BigtableOptions implements Serializable, Cloneable {
 
   private static final long serialVersionUID = 1L;
 
-  // If set to a host:port address, this environment variable will configure the client to connect
-  // to a Bigtable emulator running at the given address with plaintext negotiation.
+  // If set to a host:port address, this environment variable will configure the
```

Comment:
Please revert all of the formatting changes.  There's too much going on in this PR, so any minor refactoring should happen in a different PR.